### PR TITLE
common: add CephContext parameter to tracing::Tracer::init() in !HAVE_JAEGER branch

### PR DESCRIPTION
**Pull request by ljflores**
Followup to https://github.com/ceph/ceph/pull/50948. This wasn't originally caught since the centos 8 default build passed, but it did fail the crimson build: https://shaman.ceph.com/builds/ceph/wip-yuri3-testing-2023-08-15-0955/b5259484dbc61e8573f27e692560c6c107b9d4c1/

Fixes: https://tracker.ceph.com/issues/62761

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by ljflores** [comment]
https://shaman.ceph.com/builds/ceph/wip-lflores-testing-2-2023-09-08-1755/13c43ec225cfcbb819ed3cbc7dc7e11019b7b402/

**Comment by ljflores** [comment]
http://pulpito.front.sepia.ceph.com/lflores-2023-09-08_20:36:06-rados-wip-lflores-testing-2-2023-09-08-1755-distro-default-smithi/

**Review by rzarzynski — APPROVED** [review]
Ah, I see – the code is under [`#ifdef HAVE_JAEGER`](https://github.com/ceph/ceph/pull/53344/files#diff-b46d02199b4bb6f489ab7279d33d54db05f46245022bb937c48a94b36ba5d0cfR9).

File: src/common/tracer.h [hunk 1]
```diff
@@ -98,7 +98,7 @@ struct Tracer {
   jspan start_trace(std::string_view, bool enabled = true) { return {}; }
   jspan add_span(std::string_view, const jspan&) { return {}; }
   jspan add_span(std::string_view span_name, const jspan_context& parent_ctx) { return {}; }
-  void init(std::string_view service_name) {}
+  void init(CephContext* _cct, std::string_view service_name) {}
```
Comment (idryomov) [inline src/common/tracer.h:101]: Nit: I would suggest moving this up, before `is_enabled()` declaration, to have the stub match the real thing in `HAVE_JAEGER` branch.

File: src/common/tracer.h [hunk 2]
```diff
@@ -98,7 +98,7 @@ struct Tracer {
...
 };
   inline void encode(const jspan_context& span, bufferlist& bl, uint64_t f=0) {}
   inline void decode(jspan_context& span_ctx, ceph::buffer::list::const_iterator& bl) {}
```
Comment (idryomov) [inline src/common/tracer.h:104]: Nit: these lines are misindented, I would also add the missing blank line while at it:
```suggestion
  
inline void encode(const jspan_context& span, bufferlist& bl, uint64_t f=0) {}
inline void decode(jspan_context& span_ctx, ceph::buffer::list::const_iterator& bl) {}
```

**Review by yuvalif — APPROVED** [review]

File: src/common/tracer.h (hunk 1 above)
Comment (ljflores) [reply src/common/tracer.h:101]: Sure, I'll fix it.

File: src/common/tracer.h (hunk 2 above)
Comment (ljflores) [reply src/common/tracer.h]: Fixed

File: src/common/tracer.h [hunk 3]
```diff
@@ -94,14 +94,15 @@ class jspan {
...
   jspan start_trace(std::string_view, bool enabled = true) { return {}; }
   jspan add_span(std::string_view, const jspan&) { return {}; }
   jspan add_span(std::string_view span_name, const jspan_context& parent_ctx) { return {}; }
-  void init(std::string_view service_name) {}
 };
-  inline void encode(const jspan_context& span, bufferlist& bl, uint64_t f=0) {}
-  inline void decode(jspan_context& span_ctx, ceph::buffer::list::const_iterator& bl) {}
+
+inline void encode(const jspan_context& span, bufferlist& bl, uint64_t f=0) {}
+inline void decode(jspan_context& span_ctx, ceph::buffer::list::const_iterator& bl) {}
```
Comment (idryomov) [inline src/common/tracer.h:105]: Nit: a blank line before `}` would be nice

Comment (ljflores) [reply src/common/tracer.h:105]: Fixed

**Review by idryomov — APPROVED** [review]

**Comment by ljflores** [comment]
> http://pulpito.front.sepia.ceph.com/lflores-2023-09-08_20:36:06-rados-wip-lflores-testing-2-2023-09-08-1755-distro-default-smithi/
> 
> Running some tests here ^

Although these tests were run on an earlier SHA, the additional changes made were minor and do not warrant a huge retest IMO, especially since this is causing all crimson builds to fail.

I verified that the build compiled locally, and testing results from the earlier SHA look good: https://tracker.ceph.com/projects/rados/wiki/MAIN#Individual-testing-for-httpsgithubcomcephcephpull53344

**Pull request merged**